endpoint-cors.php

```diff
@@ -112,12 +120,17 @@ function handlePreflight() {
 }
 
 function getS3Client() {
-    global $serverPublicKey, $serverPrivateKey;
-
-    return S3Client::factory(array(
-        'key' => $serverPublicKey,
-        'secret' => $serverPrivateKey
-    ));
+    global $serverPublicKey, $serverPrivateKey, $s3Region;
+
+    return new S3Client([
+        'region' => $s3Region,
+        'version' => '2006-03-01',
+        'signature_version' => $s3Signature_version,
```

Comment:
$s3Signature_version does't defined in that scope. You should use global section for this variable